test/components/contents_list_component_test.rb

```diff
@@ -35,6 +47,23 @@ def nested_contents_list
     assert_select ".app-c-contents-list__link[href='/two']", text: "2. Two"
   end
 
+  test "renders text only when link path is nil" do
+    render_component(contents: contents_list_with_blank_path)
+    assert_select ".app-c-contents-list"
+    assert_select ".app-c-contents-list__link[href='/one']", text: "1. One"
+    assert_select ".app-c-contents-list__link[href='/two']", count: 0
```

Comment:
I feel like this line may be too specific and not actually testing what we want to test, as we're not passing a '/two' href anywhere it seems weird to check that it's not there. It may make more sense just to check that there is no link element?